### ISSUE TITLE
ideas + issues

### ISSUE THREAD

**Issue by argonsodiumvanadium** [shansitads/cbse-analyser#1]
looks great, but there are a few fixes, which i think will be appreciated 
### problems
- detail the readme with some more stuff (check out professional packages on github like [`riot search engine`](https://github.com/go-ego/riot))
- rename `controller.py` to `main.py` or `test.py`

### questions
- why is it named CBSE **analyzer**, it doesn't look like an analyzer
- why doesn't the student object use arrays, if u really want a unique name wouldn't a map suffice, as it is more iterable and versatile
- i saw that you change slice length of arrays in every program is it necessary, if yes then why ? why exactly is the code program specific

### recommendations
- in general switch to `YAML` or `JSON`, because `CSV` is just going to be irritating, i will have to write 4 -5 lines of code to slice an array, which can be done in 1 - 2 with `YAML` and `JSON` formatter, so make a function which converts all the data to `JSON` or `YAML` in the `fileReader.py` file
- just saying you should use conventions like convention in python is that any function that converts object to JSON or any format is usually called `dumps` and vice versa is called `loads`,  [this is a JSON encoder/decoder](https://docs.python.org/3/library/json.html) and [this is a yaml encoder/decoder](https://pyyaml.org/wiki/PyYAMLDocumentation)
#### i saw that you were creating a new version of the package, it would be great if you can implement this features in that version

**Comment by shansitads** [comment]
1. I had been assigned this task with some specific requirements, which I had to fulfil in a short duration so I'll agree that the code is quite messy. I have decided to make a new version for my own purpose now that the task has been completed. I'll consider these points in the new version.
2. The `CSV` output and the name of the file (cbse-analyzer) were recommended during the assignment of the task.
3. The format differs in each year, so the code was slightly different. The new version will hopefully account for those variations in a single program.
